rpcserver.go

```diff
@@ -357,12 +357,37 @@ func (r *rpcServer) MintAsset(ctx context.Context,
 	specificGroupKey := len(req.Asset.GroupKey) != 0
 	specificGroupAnchor := len(req.Asset.GroupAnchor) != 0
 
+	switch {
+	// New grouped asset and grouped asset cannot both be set.
+	case req.NewGroupedAsset && req.GroupedAsset:
+		return nil, fmt.Errorf("cannot set both new grouped asset " +
+			"and grouped asset",
+		)
+
 	// Using a specific group key or anchor implies disabling emission.
-	if req.EnableEmission {
+	case req.NewGroupedAsset:
 		if specificGroupKey || specificGroupAnchor {
 			return nil, fmt.Errorf("must disable emission to " +
 				"specify a group")
 		}
+
+	// If the asset is intended to be part of an existing group, a group key
+	// or anchor must be specified, but not both.
+	case req.GroupedAsset:
```

Comment:
Nice to see all the cases enumerated here, then compare that to the single catch all that could allow for unintended behavior.